packages/core/admin/admin/src/pages/MarketplacePage/components/NpmPackageCard/CardButton.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import semver from 'semver';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { Tooltip } from '@strapi/design-system/Tooltip';
+import { Button } from '@strapi/design-system/Button';
+import { Box } from '@strapi/design-system/Box';
+import Duplicate from '@strapi/icons/Duplicate';
+
+const TooltipButton = ({ description, installMessage, disabled, handleCopy, pluginName }) => (
+  <Tooltip data-testid={`tooltip-${pluginName}`} description={description}>
+    <Box>
+      <Button
+        size="S"
+        startIcon={<Duplicate />}
+        variant="secondary"
+        disabled={disabled}
+        onClick={handleCopy}
+      >
+        {installMessage}
+      </Button>
+    </Box>
+  </Tooltip>
+);
+
+const CardButton = ({ strapiPeerDepVersion, strapiAppVersion, handleCopy, pluginName }) => {
+  const { formatMessage } = useIntl();
+  const versionRange = semver.validRange(strapiPeerDepVersion);
+  const isCompatible = semver.satisfies(strapiAppVersion, versionRange);
+
+  const installMessage = formatMessage({
+    id: 'admin.pages.MarketPlacePage.plugin.copy',
+    defaultMessage: 'Copy install command',
+  });
+
+  // Only plugins receive a strapiAppVersion
+  if (strapiAppVersion) {
+    if (!versionRange) {
+      return (
+        <TooltipButton
+          installMessage={installMessage}
+          pluginName={pluginName}
+          description={formatMessage(
+            {
+              id: 'admin.pages.MarketPlacePage.plugin.version.null',
+              defaultMessage:
+                'Unable to verify compatibility with your Strapi version: "{strapiAppVersion}"',
+            },
+            { strapiAppVersion }
+          )}
+          handleCopy={handleCopy}
+        />
+      );
+    }
+
+    if (!isCompatible) {
+      return (
+        <TooltipButton
+          installMessage={installMessage}
+          pluginName={pluginName}
+          description={formatMessage(
+            {
+              id: 'admin.pages.MarketPlacePage.plugin.version',
+              defaultMessage:
+                'Update your Strapi version: "{strapiAppVersion}" to satisfy the compatible range: "{versionRange}"',
```

Comment:
This sounds good to me but I just want to double check that it's okay, as it's not the wording I see on Figma: "Update your Strapi version to use this [plugin/provider]"

If not maybe 1 slight change I'd recommend is saying "Update your Strapi version (strapiAppVersion) to..."